quasar/src/components/editor/QEditor.json

```diff
@@ -0,0 +1,226 @@
+    },
+    "height": {
+      "type": "String",
+      "desc": "CSS value to set the height of the input area"
+    },
+    "definitions": {
+      "type": "Object",
+      "desc": "Definition of commands and their buttons to be included in the 'toolbar' prop",
+      "examples": [
+        ":definitions=\"{",
+        "  save: {",
+        "    tip: 'Save your work',",
+        "    icon: 'save',",
+        "    label: 'Save',",
+        "    handler: saveWork",
+        "  },",
+        "  upload: {",
+        "    tip: 'Upload to cloud',",
+        "    icon: 'cloud_upload',",
+        "    label: 'Upload',",
+        "    handler: uploadIt",
+        "  }",
+        "}\""
+      ],
+      "definition": {
+        "label": {
+          "type": "String",
+          "desc": "Label of the button"
+        },
+        "tip": {
+          "type": "String",
+          "desc": "Text to be displayed as a tooltip on hover"
+        },
+        "htmlTip": {
+          "type": "String",
+          "desc": "HTML formatted text to be displayed within a tooltip on hover"
+        },
+        "icon": {
+          "type": "String",
+          "desc": "Icon of the button"
+        },
+        "key": {
+          "type": "Number",
+          "desc": "Keycode of a key to be used as a shortcut for triggering this element"
+        },
+        "handler": {
+          "type": "Function",
+          "desc": "Either this or \"cmd\" is required. Function for when button gets clicked/tapped."
+        },
+        "cmd": {
+          "type": "String",
+          "desc": "Either this or \"handler\" is required. This must be a valid execCommand method according to the designMode API."
+        },
+        "param": {
+          "type": ["String", "Function"],
+          "desc": "Only set a param if using a \"cmd\". This is commonly text or HTML to inject, but is highly dependent upon the specific cmd being called."
+        },
+        "disable": {
+          "type": ["Boolean", "Function"],
+          "desc": "Is button disabled? If specifying a function, return a Boolean value."
+        },
+        "type": {
+          "type": "String",
+          "desc": "Pass the value \"no-state\" if the button should not have an \"active\" state",
+          "values": [null, "no-state"]
+        }
+      }
+    },
+    "fonts": {
+      "type": "Object",
+      "desc": "Object with definitions of fonts",
+      "examples": [
+        ":fonts=\"{",
+        "  arial: 'Arial',",
+        "  arial_black: 'Arial Black',",
+        "  comic_sans: 'Comic Sans MS'",
+        "}\""
+      ]
+    },
+    "toolbar": {
+      "type": "Array",
+      "desc": "An array of arrays of Objects/Strings that you use to define the construction of the elements and commands available in the toolbar",
+      "default": [
+        ["left", "center", "right", "justify"],
+        ["bold", "italic", "underline", "strike"],
+        ["undo", "redo"]
+      ]
+    },
+    "toolbar-color": {
```

Comment:
"extends": "color" for all "X-color" props and "extends": "text-color" for all "X-text-color" props